### PR TITLE
feat: Added provider search_count

### PR DESCRIPTION
**Pull request by LucasTavaresA**
Now that [cmd_height](https://github.com/neovim/neovim/pull/16251) can be set to zero, i think this is a nice addition to feline
idk how to categorize this tho

### CODE REVIEW

File: doc/feline.txt [hunk 1]
```diff
@@ -858,6 +858,7 @@ Feline by default has some built-in providers to make your life easy. They are:
 │|feline-position|             │Get line and column number of cursor       │
 │line_percentage               │Current line percentage                    │
 │|feline-scroll_bar|           │Scroll bar that shows file progress        │
+│feline-search_count           │Current search total/current word count    │
```
Comment (famiu) [inline doc/feline.txt:861]: Do not modify Vim docs on your own, this will be autogenerated by a CI job. Modifying the USAGE.md file is enough

**Review by famiu — CHANGES_REQUESTED** [review]

File: lua/feline/providers/cursor.lua [hunk 2]
```diff
@@ -73,4 +73,14 @@ function M.scroll_bar(_, opts)
     end
 end
 
+function M.search_count()
+    if vim.v.hlsearch == 0 then
```
[the remark this answers is not in the document]
Comment (LucasTavaresA) [reply lua/feline/providers/cursor.lua:77]: when i check for nil the search counter never disappears even when neovim just opened or when you press `:noh`

[the remark this answers is not in the document]
Comment (famiu) [reply lua/feline/providers/cursor.lua:77]: Fair enough, I guess it's fine to keep this check. But we should probably clarify in docs that it requires `'hlsearch'`

[the remark this answers is not in the document]
Comment (LucasTavaresA) [reply lua/feline/providers/cursor.lua:77]: Just added this to `USAGE.md`

File: USAGE.md [hunk 3]
```diff
@@ -697,6 +698,10 @@ format = 'Ln {line}, Col {col}'
 
 The `scroll_bar` provider can take a `reverse` provider option, which may be either `true` or `false` and will determine if the scroll bar is reversed, which may be useful if you want the scroll bar to have natural scrolling.
 
+### Search Count
+
+Needs [hlsearch](https://neovim.io/doc/user/options.html#'hlsearch') to be able to show the count.
```
Comment (famiu) [inline USAGE.md:703]:
```suggestion
This option requires `'hlsearch'` (see `:h 'hlsearch'` in Neovim) in order for it to work.
```

**Review by famiu — APPROVED** [review]
Thanks for your contribution :)

**Pull request merged**